dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -58,6 +66,24 @@ def register_new_user(user)
     redirect_to new_user_registration_url
   end
 
+  # TODO: figure out how to avoid skipping CSRF verification for Powerschool
+  skip_before_action :verify_authenticity_token, only: :powerschool
+
+  def extract_powerschool_data(auth)
+    # OpenID 2.0 data comes back in a different format compared to most of our other oauth data.
+    args = JSON.parse(auth.extra.response.message.to_json)['args']
+    auth_info = OmniAuth::AuthHash.new(
+      user_type: args["[\"http://openid.net/srv/ax/1.0\", \"value.ext0\"]"],
+      email: args["[\"http://openid.net/srv/ax/1.0\", \"value.ext1\"]"],
+      name: {
+        first: args["[\"http://openid.net/srv/ax/1.0\", \"value.ext2\"]"],
+        last: args["[\"http://openid.net/srv/ax/1.0\", \"value.ext3\"]"],
```

Comment:
Any reason not to use outer single-quotes here, so we don't have to escape the inner double-quotes?